pkg/server/plugin/upstreamauthority/vault/vault_client.go

```diff
@@ -30,6 +31,8 @@ const (
 	defaultCertMountPoint    = "cert"
 	defaultPKIMountPoint     = "pki"
 	defaultAppRoleMountPoint = "approle"
+	defaultK8sMountPoint     = "kubernetes"
+	defaultK8sTokenPath      = "/var/run/secrets/kubernetes.io/serviceaccount/token" // #nosec G101
```

Comment:
Setting a default token path that is not a projected service account token will effectively make a choice that negatively affects the security posture, as the default service account token is unbounded.
I suggest that we should not allow leaving blank the token path assuming the default (non projected) service account token.
I recommend that instead we should require setting an explicit token path, so the choice about the token used is a conscious decision.

Reply:
That makes sense.

However, I intended as follows.
The BoundServiceAccountToken feature has been enabled (as beta) since K8s v1.21.0. It use the Projected Service Account Token as defaults and replaces the legacy Service Account Token.
Therefore, you can get the token generated by the projected service account token at the above path.

Should we take care for the Bound Service Account Token as defaults a little later?

Reply:
I think that we can evaluate that when this feature is GA (planned for v1.22), considering how many users may be running pre v1.21 deployments by that time.
My suggestion is that at that moment we can consider to introduce the default value and document this.

Reply:
I agree with your suggestion.